File: lingxia-sdk/apple/Sources/Platform/macOS/BrowserLocalNavigation.swift
#if os(macOS)
import Foundation

enum BrowserTabOpenAuthority: Equatable {
    case appSession
    case nativeControl
}

/// Browser-owned internal navigation. These routes belong to browser chrome;
/// they are not host-wide Settings destination resolution.
enum BrowserLocalNavigation: Equatable {
    case settings
    case clearSiteData(tabID: String)
    case downloads
    case bookmarks
    case history

    static let settingsTabID = "settings"

    var url: String {
        switch self {
        case .settings:
            return "lingxia://settings"
        case .clearSiteData(let tabID):
            return "lingxia://settings#clear-site-data?tabId=\(tabID)"
        case .downloads:
            return "lingxia://downloads"
        case .bookmarks:
            return "lingxia://bookmarks"
        case .history:
            return "lingxia://history"
        }
    }

    var stableTabID: String {
        switch self {
        case .settings, .clearSiteData:
            return Self.settingsTabID
        case .downloads:
            return "downloads"
        case .bookmarks:
            return "bookmarks"
        case .history:
            return "history"
        }
    }

    var openAuthority: BrowserTabOpenAuthority { .nativeControl }
}
#endif

File: crates/lingxia-windows-sdk/src/browser_local_navigation.rs
/// Browser-owned management navigation. These routes are intentionally
/// separate from the product-wide static Settings destination resolver.
#[derive(Debug, Clone, PartialEq, Eq)]
pub(crate) enum BrowserLocalNavigation<'a> {
    Settings,
    ClearSiteData { tab_id: &'a str },
}

impl BrowserLocalNavigation<'_> {
    pub(crate) fn url(&self) -> String {
        match self {
            Self::Settings => "lingxia://settings".to_string(),
            Self::ClearSiteData { tab_id } => {
                format!("lingxia://settings#clear-site-data?tabId={tab_id}")
            }
        }
    }
}

#[cfg(test)]
mod tests {
    use super::*;

    #[test]
    fn browser_local_settings_and_clear_site_data_keep_private_routes() {
        assert_eq!(BrowserLocalNavigation::Settings.url(), "lingxia://settings");
        assert_eq!(
            BrowserLocalNavigation::ClearSiteData { tab_id: "tab-1" }.url(),
            "lingxia://settings#clear-site-data?tabId=tab-1"
        );
    }
}
